### PR TITLE
Improve the Curie build file generator a bit

### PR DESCRIPTION
**Pull request by galpeter**
Works with python 2 and 3.

### CODE REVIEW

**Comment by zherczeg** [comment]
@jiangzidong could you try this patch?

File: targets/curie_bsp/setup.py [hunk 1]
```diff
@@ -14,138 +16,147 @@
...
-    out_put = '''
+def build_soft_links(project_path, jerry_path):
+    """ Creates soft links into the @project_path. """
+
+    if not os.path.exists(project_path):
+        os.makedirs(project_path)
+
+    links = [
+        { # arc
+            'src': 'targets/curie_bsp/jerry_app/arc',
```
Comment (akosthekiss) [inline targets/curie_bsp/setup.py]: Why no `os.path.join()` for 'src'?

File: targets/curie_bsp/setup.py [hunk 2]
```diff
@@ -14,138 +16,147 @@
...
+        src = os.path.join(jerry_path, link['src'])
+        link_name = os.path.join(project_path, link['link_name'])
+        if not os.path.islink(link_name):
+            os.symlink(src, link_name)
+            print("Created symlink '{link_name}' -> '{src}'".format(src=src, link_name=link_name))
+
+
+def find_sources(root_dir, sub_dir):
+    """
+    Find .c and .S file inside the @root_dir/@sub_dir directory.
```
Comment (akosthekiss) [inline targets/curie_bsp/setup.py]: (typo) file -> files

File: targets/curie_bsp/setup.py (hunk 1 above)
Comment (galpeter) [inline targets/curie_bsp/setup.py]: Good question! IMHO it's readable, and now I noticed that below there is a join for 'quark' and 'include' (also there is not variable part in the path here and if for whatever reason you want to run this on windows the `/` paths will work)

File: targets/curie_bsp/setup.py [hunk 3]
```diff
@@ -159,62 +170,83 @@ def create_makefile_in_curie(curie_path):
...
-        print 'Usage:'
-        print sys.argv[0] + ' [full or relative path of Curie_BSP]'
-    else:
-        file_dir = os.path.dirname(os.path.abspath(__file__))
-        jerry_path = os.path.join(file_dir, "..", "..")
-        curie_path = os.path.join(os.getcwd(), sys.argv[1])
-        main(curie_path, jerry_path)
+    import sys
+
+    if len(sys.argv) < 2:
```
Comment (akosthekiss) [inline targets/curie_bsp/setup.py]: Why the change from `!=` to `<`? What would be the meaning of accepted optional command line arguments in `sys.argv[2:]`?

Comment (galpeter) [inline targets/curie_bsp/setup.py]: Just wanted to be a bit more fault tolerant (accidentally putting two paths there :)) but you are right it would be better to leave it as `!=`

**Comment by galpeter** [comment]
@akosthekiss updated based on your comments.

**Comment by jiangzidong** [comment]
@zherczeg sure, I will try the patch on device next morning :)

**Comment by LaszloLango** [comment]
@galpeter, please fix the Sign-off in the commit message

**Comment by galpeter** [comment]
ohh... yeah forgot that thing.

File: targets/curie_bsp/setup.py [hunk 4]
```diff
@@ -159,62 +170,83 @@ def create_makefile_in_curie(curie_path):
...
+    quark_include_paths = [
+        'include',
+        'jerryscript',
+        'jerryscript/jerry-libm/include',
+        'jerryscript/targets/curie_bsp/include'
+    ] + list(jerry_data['dirs'])
+
+    quark_includes = [
+        '-Wno-error',
+    ] + ['-I$(PROJECT_PATH)/quark/%s' % dir for dir in quark_include_paths]
```
Comment (akosthekiss) [inline targets/curie_bsp/setup.py]: Do we need `$(PROJECT_PATH)` here or do we have that already from the `project_path` variable?
And why not `['-I%s' % os.path.join(project_path, 'quark', dir) for dir in quark_include_paths]`?

File: targets/curie_bsp/setup.py [hunk 5]
```diff
@@ -159,62 +170,83 @@ def create_makefile_in_curie(curie_path):
...
+    jerry_data = build_jerry_data(jerry_path)
+    jerry_objects = build_obj_y(jerry_data['sources'])
+    jerry_defines = jerry_data['cflags']
+    jerry_build_dirs = build_mkdir(jerry_data['dirs'])
+
+    quark_include_paths = [
+        'include',
+        'jerryscript',
+        'jerryscript/jerry-libm/include',
+        'jerryscript/targets/curie_bsp/include'
```
Comment (akosthekiss) [inline targets/curie_bsp/setup.py]: If we use `os.path.join` everywhere, might be nice to use it here as well.

File: targets/curie_bsp/setup.py (hunk 4 above)
Comment (galpeter) [inline targets/curie_bsp/setup.py]: I wanted to keep the generated code as close as possible to the previous version. Also I don't know if there would be any side effect if we change it to the `project_path` one.

Comment (akosthekiss) [inline targets/curie_bsp/setup.py]: might be good to hear @jiangzidong 's opinion

Comment (jiangzidong) [inline targets/curie_bsp/setup.py]: no side effect here. it is ok to replace $(PROJECT_PATH)

**Comment by jiangzidong** [comment]
@galpeter the patch works well on real device (and I also tested the` ['-I%s' % os.path.join(project_path, 'quark', dir) for dir in quark_include_paths]`)
LGTM, thanks!

**Comment by galpeter** [comment]
@jiangzidong that's good to hear. I'll do a final round of update.

**Comment by galpeter** [comment]
@akosthekiss everything should be fixed now. PTAL.

**Comment by akosthekiss** [comment]
LGTM

**Comment by LaszloLango** [comment]
LGTM

**Comment by akosthekiss** [comment]
Please, update to latest master to get landed

**Comment by galpeter** [comment]
rebased to the latest master.
